contracts/modules/capital/SwapOperator.sol

```diff
@@ -295,6 +300,81 @@ contract SwapOperator is ReentrancyGuard {
     emit Swapped(ETH, stETH, amountIn, amountOut);
   }
 
+  function swapETHForEnzymeVaultShare(uint amountIn, uint amountOutMin) external onlySwapController {
+    IPool pool = _pool();
+    IEnzymeV4Comptroller comptrollerProxy = IEnzymeV4Comptroller(IEnzymeV4Vault(enzymeV4VaultProxyAddress).getAccessor());
+    address toTokenAddress = enzymeV4VaultProxyAddress;
+    address weth = router.WETH();
+
+    (
+    uint112 minAmount,
+    uint112 maxAmount,
+    /* uint32 lastAssetSwapTime */,
+    /* uint maxSlippageRatio */
+    ) = pool.getAssetDetails(toTokenAddress);
+
+    require(!(minAmount == 0 && maxAmount == 0), "SwapOperator: asset is not enabled");
+    
+    uint balanceBefore = IERC20(toTokenAddress).balanceOf(address(pool));
+
+    pool.transferAssetToSwapOperator(ETH, amountIn);
+
+    require(comptrollerProxy.getDenominationAsset() == weth, "SwapOperator: invalid denomination asset");
+
+    enzymeV4DepositWrapper.exchangeEthAndBuyShares{value: amountIn}(address(comptrollerProxy), weth, 1, address(0), address(0), '0x', 0);
+
+    pool.setAssetDataLastSwapTime(toTokenAddress, uint32(block.timestamp));
+
+    uint amountOut = IERC20(toTokenAddress).balanceOf(address(this));
+
+    require(amountOut >= amountOutMin, "SwapOperator: amountOut < amountOutMin");
```

Comment:
to confirm is there potential slippage here in terms of how many shares you get from your ETH deposit?

Example:  i deposit 1 ETH, i may get a variable number of shares depending on when i deposited.

Reply:
Yes. It'll depend on the price of the underlying basket of assets. So it depends on the block number